datahub-web-react/src/app/entity/shared/containers/profile/header/PlatformContent/DatasetLink.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+import styled from 'styled-components';
+import { Link } from 'react-router-dom';
+import { Typography } from 'antd';
+import { Dataset, EntityType } from '../../../../../../../types.generated';
+import { ANTD_GRAY } from '../../../../constants';
+import { useEntityRegistry } from '../../../../../../useEntityRegistry';
+
+const DatasetText = styled(Typography.Text)`
+    font-size: 12px;
+    line-height: 20px;
+    color: ${ANTD_GRAY[7]};
+`;
+
+export const DatasetIcon = styled.span`
+    color: ${ANTD_GRAY[7]};
+    margin-right: 4px;
+    font-size: 12px;
+`;
+
+const StyledLink = styled(Link)`
+    white-space: nowrap;
+`;
+
+interface Props {
+    parentDataset: Dataset;
+    parentDatasetIcon: JSX.Element;
+}
+
+function DatasetLink(props: Props) {
+    const { parentDataset, parentDatasetIcon } = props;
+    const entityRegistry = useEntityRegistry();
+    if (!parentDataset) return null;
+
+    const datasetUrl = entityRegistry.getEntityUrl(EntityType.Dataset, parentDataset.urn);
+    const datasetName = entityRegistry.getDisplayName(EntityType.Dataset, parentDataset);
+
+    return (
+        <StyledLink to={datasetUrl} data-testid="dataset">
+            <DatasetIcon>{parentDatasetIcon}</DatasetIcon>
```

Comment:
i think passing `parentDatasetIcon` down these levels is unnecessary since it should always just be a regular dataset icon you get by using `entityRegistry.getIcon(EntityType.Dataset, 14, IconStyleType.ACCENT)` we could just get the icon here in this component and if we ever want to start providing dynamic icons we can accept that as a prop for this component

Reply:
Agreed, I have made the changes suggested above. Thanks!